### DISCUSSION TITLE
Python Agents

### DISCUSSION THREAD

**Discussion by corytomlinson** [microsoft/semantic-kernel#8278] [General]
Trying to use agents with OpenAI Assistants in python for the first time after reading https://devblogs.microsoft.com/semantic-kernel/introducing-agents-in-semantic-kernel/ and getting nowhere.

1. Code snippets are incomplete (don't include required imports)
2. Appears to be no documentation for the new functionality

To the post authors, this kind of this is not helpful at all -> "Once you’ve installed and **imported the right packages**, you can quickly get started with just a few lines of code:" Just include the imports in the snippet so people can easily get started.

**Comment by corytomlinson** [comment]
Managed to find 2...

```
from semantic_kernel.contents.utils.author_role import AuthorRole
from semantic_kernel.contents.chat_message_content import ChatMessageContent
```

These others are proving to be more difficult to locate...

```
AzureAssistantAgent
display_message
```

**Comment by corytomlinson** [comment]
`from semantic_kernel.agents.open_ai.azure_assistant_agent import AzureAssistantAgent`

**Comment by corytomlinson** [comment]
After sorting out the imports the code now runs but produces an error...

`agent = await AzureAssistantAgent.create` results in `NotFoundError: Error code: 404 - {'error': {'code': '404', 'message': 'Resource not found'}}`

Is there some other initialization code that was left out of the example that is required to make this work?

**Comment by corytomlinson** [comment]
There must be a bug in OpenAI `1.40.8` that was causing the `404`. Have installed `1.41.0` and it is working now. For anyone else struggling to get this example working the imports needed are...

```
from semantic_kernel.contents.utils.author_role import AuthorRole
from semantic_kernel.contents.chat_message_content import ChatMessageContent
from semantic_kernel.agents.open_ai.azure_assistant_agent import AzureAssistantAgent
```

I can only assume that `display_message` is some custom function that is not part of the library as I cannot find it anywhere.

**Comment by moonbox3** [comment]
Hi @corytomlinson, thank you for your feedback. Apologies that it's been difficult to get up and running. I'll work on getting the blog updated with the required imports/methods (and we'll include these going forward), as well as links to our other agent samples:

1. [Getting started with agents](https://github.com/microsoft/semantic-kernel/tree/main/python/samples/getting_started_with_agents)
2. [Agent concept samples](https://github.com/microsoft/semantic-kernel/tree/main/python/samples/concepts/agents)

Please note that you'll need your .env file configured as mentioned [here](https://github.com/microsoft/semantic-kernel/blob/main/python/samples/concepts/README.md#steps-for-configuration). Otherwise, you can directly pass in the api_key/deployment_name/endpoint.

**Comment by moonbox3** [comment]
As a follow-up, @corytomlinson, I've updated the blog so that each code snippet (whether in C# or Python) then points to the full implementation of that scenario. For example, the "OpenAI Assistants" code snippet has a link below it to point the reader to the sample:

![image](https://github.com/user-attachments/assets/1f36656c-b05c-465c-ae1b-8d4cfdd41f89)


https://devblogs.microsoft.com/semantic-kernel/introducing-agents-in-semantic-kernel/

Thanks again for your feedback, we appreciate it!
